testsuite/Makefile

```diff
@@ -29,9 +29,16 @@ TESTLOG ?= _log
 
 ocamltest_directory := ../ocamltest
 
+ifeq "$(NATIVE_COMPILER)" "true"
+  ocamltest_program_to_build := ocamltest.opt$(EXE)
+else
+  ocamltest_program_to_build := ocamltest$(EXE)
+endif
+
 ocamltest_program := $(or \
   $(wildcard $(ocamltest_directory)/ocamltest.opt$(EXE)),\
-  $(wildcard $(ocamltest_directory)/ocamltest$(EXE)))
+  $(wildcard $(ocamltest_directory)/ocamltest$(EXE)),\
+  $(ocamltest_directory)/$(ocamltest_program_to_build))
```

Comment:
Idle question: if we had had a third `or` argument being `$(error ocamltest not found ..)` before, would that have been equivalent to the `ifeq "$(ocamltest_program)" ""` condition below?

Reply:
Not quite - it's the difference `:=` (evaluate immediately and once) and `=` (evaluate lazily)!